### PR TITLE
Initial overview documentation of protocols

### PR DESCRIPTION
**Pull request by mpharrigan**
This is a step towards #503.

As part of writing full docs for decompose, classical sim, cirq interop, drawing, and other protocols: their corresponding section should be fleshed out in this overview page.

### CODE REVIEW

File: qualtran/Protocols.ipynb [hunk 1]
```diff
@@ -0,0 +1,72 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "a3bea287",
+   "metadata": {},
+   "source": [
+    "# Protocols\n",
```
Comment (NoureldinYosri) [inline qualtran/Protocols.ipynb:8]: If I didn't know what a protocol is, I wouldn't have understood this section. I think a simple paragraph that doesn't mention bloqs and references some good explanation should make this section more understandable.

Comment (mpharrigan) [reply qualtran/Protocols.ipynb:8]: well I don't think we're using the word "protocol" accurately in an established way:

https://en.wikipedia.org/wiki/Protocol

but rather leaning in to its ambiguous meaning (https://en.wiktionary.org/wiki/protocol) to define it exactly how we want; i.e. below.

**Review by fdmalone — APPROVED** [review]

File: qualtran/Protocols.ipynb (hunk 1 above)
Comment (fdmalone) [reply qualtran/Protocols.ipynb:8]: To be clear is this the idiomatic meaning of protocol and not any formal CS protocol type thing

Comment (mpharrigan) [reply qualtran/Protocols.ipynb:8]: how about this updated wording:

> We write bloqs so we can query properties, draw diagrams, and check the correctness of quantum algorithms. Basically: we write bloqs so we can do things with them! Each 'thing'--or genre of thing--we want to do has an associated "protocol". In our usage of the term, a protocol is a set of methods, functions, or procedures to structure a relevant piece of functionality.

**Pull request merged**